src/handlers/authenticationHandlers.js

```diff
@@ -1,67 +1,94 @@
 import axios from 'axios';
 
-import { pathObj } from './../utils/generalVariables';
-import { setHeaders } from './../utils/requestHeaders'
+import { pathObj, tokenKey } from './../utils/generalVariables';
+import { setHeaders } from './../utils/requestHeaders';
 
 export const loginHandler = ({ username, password }) => {
   // With the finalization of the database schema, more checks can be implemented (with separate error-messages)
-  if (!username || !password || typeof username !== "string" || typeof password !== "string" || password.length < 6 ) {
-    return new Error("Make sure you're passing a valid username and a password that's at least 8 characters long")
+  if (
+    !username ||
+    !password ||
+    typeof username !== 'string' ||
+    typeof password !== 'string' ||
+    password.length < 6
+  ) {
+    return new Error(
+      "Make sure you're passing a valid username and a password that's at least 8 characters long"
+    );
   }
 
-  return axios.post(`${pathObj.loginPath}`,  { 
-    username, 
+  return axios.post(`${pathObj.loginPath}`,  {
+    username,
     password,
   })
     .then(res => {
-      if (res.token) {
+      if (res.data.token) {
         checkAndStoreToken(res.data.token);
-        return res.data.user
+        return res.data.user;
       } else {
-        return new Error('Oh no, there was no token returned by the database!')
+        return new Error('Oh no, there was no token returned by the database!');
```

Comment:
no throw here?

Reply:
Okaaay, I was trying to leave this because this PR has already a monstruous scope and I've stepped into parts that I should haven't, but I'll change them from here.